### DISCUSSION TITLE
The Nations Gold Edition by JoWood game fully functional with dxwrapper!

### DISCUSSION THREAD

**Discussion by axtox** [elishacloud/dxwrapper#271] [Show and tell]
# Intro
I really love this game so I decided to make it work with Windows 10 no matter what. I've tried different PC's, different GPU's and even Virtual Machines. And finally I was able to run it. 

In my case the game was pretty much functional, thanks to root folder installation, but the mouse cursor was blinking as well as screen during the game - so I solved that using dxwrapper.

![The_Nations_2001_Cover](https://github.com/elishacloud/dxwrapper/assets/11292461/e5fe1c0b-0c24-406c-a2d8-031be2a90284)

# Setup
- Windows 10 x64
- Nvidia GeForce GTX 1050 ti
- LG with 3440x1440 px
- Game installed on root "C:\" drive folder (C:\ZM4 in my case)
- Latest ([v1.1.6900.22](https://github.com/elishacloud/dxwrapper/releases/tag/v1.1.6900.22)) dxwrapper unpacked into /bin folder (C:\ZM4\bin in my case)

# How To:
## Game Installation
First of all the game should be installed on the root drive which is C:\ in my case. During the installation you have to change the installation folder to any folder exactly at C:\ drive - I guess this is due to unhandled spaces or symbols in the Windows 10 paths that occurs on x64 when installing x32 apps (they goes into C:\Program Files (x32)\ folder).
> [!WARNING]
> If you skip this step you'll see the black screen every time trying start new game. In some cases you'll hear the sea sounds but that's all

## Game Settings
Once installed locate the installation folder - there will be a `\tools` folder which contains `setRegTrouble.exe`, sometimes it is called little bit different, just try to find some kind of a settings executable. It looks somewhat like this:

![изображение](https://github.com/elishacloud/dxwrapper/assets/11292461/c063247c-60c3-4b68-b43e-3be88837c593)

Set all the settings just like in the screenshot:
1. Choose `custom` option from dropdown
2. **Disable** the AGP-texturing using dropdown
3. Set "Texture Restriction" to **4000**
4. Click on **Resolution** checkbox and allow to choose in-game resolution

## DxWrapper Installation
There's no installation actually, just download `dxwrapper.zip` file from the [latest release](https://github.com/elishacloud/dxwrapper/releases/) from Release Assets and unpack it right inside the `\bin` folder in game installation folder (`C:\ZM4\bin` in my case)

## DxWrapper Settings
> [!NOTE]
> There is a lot of settings in the dxwrapper, so if I did not mention some of them, this means that you should leave this setting as is with no changes!

I noticed that this game works best with `ddrawcompact`, so the settings are mostly related to this technology.  
This is the most important step - it wont run without these settings, there will be a lot of artifacts if you leave it as it is, so here's the correct settings for dxwrapper:
1. Open `dxwrapper.ini` file with any text editor
2. Search for `Compatibility` setting group and set this settings to `1`:
```env
DDrawCompat                = 1
Dinputto8                  = 1
EnableDdrawWrapper         = 1
EnableD3d9Wrapper          = 1
EnableDinput8Wrapper       = 1
DisableHighDPIScaling      = 1
```
3. Search for `DDrawCompat` setting group and set this settings to `1`:
```env
DDrawCompat31              = 1
```
4. Search for `ddraw` setting group and set this settings to `1`:
```env
ConvertToDirectDraw7       = 1
ConvertToDirect3D7         = 1
```
5. Search for `d3d9` setting group and set this settings to `1`:
```env
ForceDirect3D9On12         = 1
GraphicsHybridAdapter      = 1
```

All other settings should be set to default values

# Enjoy!
You now can just open the game and play! The resolution settings should work fine - but stick to 16:9 ratio, on my 21:9 the maximum possible resolution was 1260x920.

Good luck!

**Comment by Dralon98** [comment]
Hello :) i have the original game in spanish, do you think you can make it work if i give you the ISO image of the game?

**Comment by elishacloud** [comment]
I am using the GOG version of the game, but I tested it and it works with the latest version of dxwrapper.  Just unzip the dx7 binary files into the "bin" folder of the game.

_Edit: the only option you need to enable is `Dd7to9`.  You can get the latest dx7 binary files from the [Releases](https://github.com/elishacloud/dxwrapper/releases) page._

**Reply by Dralon98** [reply]
I just followed all of the steps above but it's not running :( 

I must say i have the original cd of this game (not the Gold Edition), that's how i made the iso copy, and that game could run even in windows 7 SO, after windows 8 i can't play it anymore. Also, this version doesn't seem to have a tools folder or any SetRegTrouble.exe file in it, that seems to be the case with the original version archive (game is in english) i found in the Internet Archive site.

So, please, if you could, i will forever be thankful :(

**Reply by elishacloud** [reply]
Can you send me your logs?  It should be called:  dxwrapper-game.log

_Edit: if you don't have a bin folder then just put the three files from the zip download into the whatever folder has the "game.exe" file in it._

**Reply by Dralon98** [reply]
I can't find a dxwrapper-game.log file :0 

(What i meant and sorry for my bad english)

When i install the version i have the program just doesn't start it just stays in the window that let's you select "play game, install game, uninstall, and so on". This is the version i refer when saying that after installation in the program files "The Nations" folder there is no folder with the name tools, i guess that is only in the GOG version.

This is the bin folder i have after unzipping DxWrapper:

![image](https://github.com/user-attachments/assets/18a22de2-4f1a-4dfc-ac05-d322d24068c3)

**Reply by elishacloud** [reply]
Ok, I see the issue.

1. Delete the following files: `dxwrapper.asi`, `AllSettings.ini` and delete the folder `Stub`.
2. Unzip these files into "The Nations" folder:  [dxwrapper.zip](https://github.com/user-attachments/files/17670703/dxwrapper.zip)

**Reply by Dralon98** [reply]
Didn't work :(

**Reply by elishacloud** [reply]
Is there a log file now?  What files do you have in that folder now?

**Reply by Dralon98** [reply]
![image](https://github.com/user-attachments/assets/44b91ff7-b21d-438c-b8f4-e2a670412436)


Same files, no log file yet.

![image](https://github.com/user-attachments/assets/79a2a160-34a5-4dfe-b655-c55408b78988)

This is the only window that i can see, after clicking play game the window hides itself and pops up inmediatly after. So guessing the game isn't even loading.

**Reply by elishacloud** [reply]
Try running the "game.exe" file directly by double clicking on it.

**Reply by Dralon98** [reply]
Already tried it, with no results :(

**Reply by elishacloud** [reply]
If it is not creating any log file then it means dxwrapper is not running.

1. What is the full path to the game?
2. What kind of system do you have?  OS?  CPU?

**Reply by Dralon98** [reply]
OS: Windows 10
Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz

The path to the .exe file: C:\Program Files (x86)\JoWooD\TheNations\bin

**Reply by elishacloud** [reply]
Can you give me a copy of your game.exe file?

**Reply by Dralon98** [reply]
Yeah, just the game.exe : 
[game.zip](https://github.com/user-attachments/files/17671025/game.zip)

**Reply by elishacloud** [reply]
The crash happens before the game even loads ddraw.dll.  I was able to get it to load dxwrapper by renaming ddraw.dll to winmm.dll.  However, it still crashes before loading ddraw.  This is an issue with the game and would need someone to create a specific patch for this game.

You could try with my version.  Make sure to backup your version and then you can download this and unzip it to the "bin" folder:  [game.zip](https://github.com/user-attachments/files/17673077/game.zip)

**Reply by axtox** [reply]
Glad to hear that this tutorial actually was proven my mainteners, maybe it's time to update the [Wiki Page](https://github.com/elishacloud/dxwrapper/wiki/DirectDraw-to-Direct3D9-Conversion#supported-2d-games-directdraw) on this game compatability?

**Reply by elishacloud** [reply]
> Glad to hear that his tutorial actually was proven my mainteners, maybe it's time to update the [Wiki Page](https://github.com/elishacloud/dxwrapper/wiki/DirectDraw-to-Direct3D9-Conversion#supported-2d-games-directdraw) on this game compatability?

Done!

**Reply by Kupo91** [reply]
> The crash happens before the game even loads ddraw.dll. I was able to get it to load dxwrapper by renaming ddraw.dll to winmm.dll. However, it still crashes before loading ddraw. This is an issue with the game and would need someone to create a specific patch for this game.
> 
> You could try with my version. Make sure to backup your version and then you can download this and unzip it to the "bin" folder: [game.zip](https://github.com/user-attachments/files/17673077/game.zip)

Hello elisha, I also try to get running the non-gold version of the game, since I always struggled with the infinite child born bug in the gold version. But I also have issues to get it running. By default nothing happens if I try to start the game. Then I placed the 3 dx files from your zip into the bin folder. After that it told me to insert the disc. But mounting the iso didn't help. Then I replaced my game.exe by yours and got this error:

![Unbenannt](https://github.com/user-attachments/assets/684d439b-fe44-4351-8823-0c5124ba649a)

Any ideas what I could do at this point?

Thanks

PS: With your exe dxwrapper created a log file: 
[dxwrapper-game.log](https://github.com/user-attachments/files/18718007/dxwrapper-game.log)

PS2: If I rename ddraw to winmm, it also creates a log file with my own exe:
[dxwrapper-game.log](https://github.com/user-attachments/files/18718027/dxwrapper-game.log)

PS3: I was also wondering, if the DX7 wrapper is really the right one, because the installer asks me for DX8 ?
![Unbenannt](https://github.com/user-attachments/assets/830f51b4-c734-4cab-8081-1c1108b46cb8)

**Reply by axtox** [reply]
> Then I placed the 3 dx files from your zip into the bin folder. After that it told me to insert the disc. But mounting the iso didn't help. 
> Any ideas what I could do at this point?

I will add my 5cents - ensure that the game is installed in the root directory of the `C:/`, for example - `C:/games` or `C:/nations`, also ensure that there's no space or any other special symbols in the name of the path
Next, be sure that you've mounted ISO file with the game disk to the first drive, meaning that, if you have only one hard drive called `C:/`, the disk drive should be `D:/` and no other disk drives exists in the system. 
I had a problem using two disk drives and especially when I mount the ISO file to the second one.

Lastly, try all the steps from my tutorial in the topic with all the settings and try not to replace `.exe` file with those that were attached here.
